spec/bundler/source_spec.rb

```diff
@@ -0,0 +1,118 @@
+    end
+
+    it "should return the names of unmet dependencies" do
+      expect(subject.unmet_deps).to eq(unmet_dependency_names)
+    end
+  end
+
+  describe "#version_message" do
+    let(:spec) { double(:spec, :name => "nokogiri", :version => ">= 1.6") }
+
+    shared_examples_for "the lockfile specs are not relevant" do
+      it "should return a string with the spec name and version" do
+        expect(subject.version_message(spec)).to eq("nokogiri >= 1.6")
+      end
+    end
+
+    context "when there are locked gems" do
+      let(:locked_gems) { double(:locked_gems) }
+
+      before { allow(Bundler).to receive(:locked_gems).and_return(locked_gems) }
+
+      context "that contain the relevant gem spec" do
+        before do
+          specs = double(:specs)
+          allow(locked_gems).to receive(:specs).and_return(specs)
+          allow(specs).to receive(:find).and_return(locked_gem)
+        end
+
+        context "without a version" do
+          let(:locked_gem) { double(:locked_gem, :name => "nokogiri", :version => nil) }
+
+          it_behaves_like "the lockfile specs are not relevant"
+        end
+
+        context "with the same version" do
+          let(:locked_gem) { double(:locked_gem, :name => "nokogiri", :version => ">= 1.6") }
+
+          it_behaves_like "the lockfile specs are not relevant"
+        end
+
+        context "with a different version" do
+          let(:locked_gem) { double(:locked_gem, :name => "nokogiri", :version => "< 1.5") }
+
+          it "should return a string with the spec name and version and locked spec version" do
+            expect(subject.version_message(spec)).to eq("nokogiri >= 1.6 (was < 1.5)")
+          end
+        end
+      end
+
+      context "that do not contain the relevant gem spec" do
+        before do
+          specs = double(:specs)
+          allow(locked_gems).to receive(:specs).and_return(specs)
+          allow(specs).to receive(:find).and_return(nil)
+        end
+
+        it_behaves_like "the lockfile specs are not relevant"
+      end
+    end
+
+    context "when there are no locked gems" do
+      before { allow(Bundler).to receive(:locked_gems).and_return(nil) }
+
+      it_behaves_like "the lockfile specs are not relevant"
+    end
+  end
+
+  describe "#can_lock?" do
+    context "when the passed spec's source is equivalent" do
+      let(:spec) { double(:spec, :source => subject) }
+
+      it "should return true" do
+        expect(subject.can_lock? spec).to eq(true)
+      end
+    end
+
+    context "when the passed spec's source is not equivalent" do
+      let(:spec) { double(:spec, :source => double(:other_source)) }
+
+      it "should return false" do
+        expect(subject.can_lock? spec).to eq(false)
+      end
+    end
+  end
+
+  describe "#include?" do
+    context "when the passed source is equivalent" do
+      let(:source) { subject }
+
+      it "should return true" do
+        expect(subject.include? source).to eq(true)
```

Comment:
Another instance of rspec using a defined method/operation (`include?`) under the hood?